Issue 6 - Filter and Sorting for tables

devmarcstorm opened the pull request:
Tabellen wie in users oder groups können nun über ein Textfield gefiltert werden.
Sortierungen durch das Anklicken von headern  ist nun auch möglich.

Beides clientseitig mit javascript.

Issue #6

devmarcstorm committed:
Add alias iteration in user template

devmarcstorm committed:
Create alias-list for user

devmarcstorm committed:
Add list filter functionality

devmarcstorm committed:
Add filter functionality to all list views

devmarcstorm committed:
Add sorting functionality to tables in list views

devmarcstorm committed:
Add translation

rixx commented on ode/blueprints/amu/static/amu/table_functions.js:
```diff
@@ -0,0 +1,86 @@
+function sortTable(n, table_index) {
+    var table = document.getElementsByTagName("table")[table_index]
```
Hier wäre es eigentlich geschickter, den Tabellen eine ID zu geben, und statt `table_index` die ID zu übergeben, und dann mit `getElementByID` zu nutzen. Dann ist es leichter, mal eine Tabelle durch die Gegend zu bewegen, eine höher einzuschieben, etc.

rixx commented on ode/blueprints/amu/static/amu/table_functions.js:
```diff
@@ -0,0 +1,86 @@
+        var rows = table.rows;
+        var canSwitch;
+        var i;
+        for (i = 1; i < (rows.length - 1); i++) {
```
Das liest sich relativ umständlich – ich würde da eher direkt auf den eingebauten Sortiermechanismus von JavaScript zurückgreifen. [Hier](https://stackoverflow.com/questions/14267781/sorting-html-table-with-javascript) ist ein ganz nettes Beispiel dafür.

rixx commented on ode/blueprints/amu/static/amu/table_functions.js:
```diff
@@ -0,0 +1,86 @@
+    }
+}
+
+function searchTable() {
```
Hier wäre es vermutlich auch gut, eine ID zu übergeben, damit man auf einer Seite auch mehrere Tabellen durchsuchen kann.

rixx commented on ode/blueprints/amu/templates/amu/aliases.html:
```diff
@@ -2,22 +2,33 @@
+  <thead>
+    <tr class="no_filter" style="cursor: pointer;">
+      <th onclick="sortTable(0, 0)">{{_("Name")}}</th>
+      <th onclick="sortTable(0, 0)">{{_("Members")}}</th>
```
Wäre das nicht `sortTable(1, 0)`?

rixx commented on ode/blueprints/amu/templates/amu/birthdays.html:
```diff
@@ -15,13 +15,18 @@ <h2>{{_("Birthdays")}}</h2>
 </ul>
 
-<table class="table table-striped">
+<div class="form-group" style="margin-bottom: 20px;">
```
Ich bin gerade in ODE nicht so drin: Eigentlich nutzen wir hier keine inline-Styles, oder? Du könntest da ja eine Klasse `filter-form` o.ä. für nehmen.

devmarcstorm replied:
Ich habe den Style komplett entfernt. Er war nicht wirklich notwendig.

rixx commented on ode/blueprints/amu/templates/amu/birthdays.html:
```diff
@@ -15,13 +15,18 @@ <h2>{{_("Birthdays")}}</h2>
-<table class="table table-striped">
+<div class="form-group" style="margin-bottom: 20px;">
+  <label>{{_("Filter list")}}:</label>
+  <input id='search_input' onkeyup='searchTable()' type="text" class="form-control" id="usr" autofocus>
```
Der Input hat zweimal eine `id`.

rixx commented on ode/blueprints/amu/templates/amu/groups.html:
```diff
@@ -2,24 +2,35 @@
+
+<div class="form-group" style="margin-bottom: 20px;">
+  <label>{{_("Filter list")}}:</label>
+  <input id='search_input' onkeyup='searchTable()' type="text" class="form-control" id="usr" autofocus>
```
Der Input hat auch zwei `id`s.

rixx commented on ode/blueprints/amu/views.py:
```diff
@@ -148,7 +148,22 @@ def user(uid):
+			if member.split(",")[1][3:] == "Users":
+				member_list.append(str(member.split(",")[0][4:]))
+		alias_dict[alias] = member_list
+	print(alias_dict)
```
Bitte noch das Print-Statement raus.

devmarcstorm replied:
Doch noch eins vergessen :)

devmarcstorm committed:
Make changes according to the Feedback to Pull Request 23

1. Another sorting algorithm implemented
 - The previously used algorithm was too cumbersome
 - The example used can be found here: https://stackoverflow.com/questions/14267781/sorting-html-table-with-javascript

2. Also, filtering tables now uses IDs
 - This allows multiple tables to be filtered on one page

3. Inline style has been removed
 - It was not needed and also not outsourced to an external css

4. The inputs had 2 IDs. One was not used, I removed this one

5. In views.py I have removed a print statement
 - Was only used for debugging and forgotten when pushing
